src/main/kotlin/org/rust/lang/core/types/RustEnumType.kt

```diff
@@ -4,12 +4,21 @@ import com.intellij.codeInsight.completion.CompletionUtil
 import org.rust.lang.core.psi.RustEnumItemElement
 import org.rust.lang.core.types.visitors.RustTypeVisitor
 
-class RustEnumType(enum: RustEnumItemElement) : RustStructOrEnumTypeBase() {
+class RustEnumType(
+    enum: RustEnumItemElement,
+    typeArguments: List<RustType> = emptyList()
+) : RustStructOrEnumTypeBase(typeArguments) {
```

Comment:
I'll try the decorator solution, but I worry that it may break (for good or for bad) matching over types like

```
when (fooType) {
    is RustStructType -> ....
}
```